client/state/purchases/actions.js

```diff
@@ -87,30 +89,7 @@ export const fetchSitePurchases = siteId => dispatch => {
 		} );
 };
 
-export const fetchUserPurchases = userId => dispatch => {
-	dispatch( {
-		type: PURCHASES_USER_FETCH,
-	} );
-
-	return new Promise( ( resolve, reject ) => {
-		wpcom.me().purchases( ( error, data ) => {
-			error ? reject( error ) : resolve( data );
-		} );
-	} )
-		.then( data => {
-			dispatch( {
-				type: PURCHASES_USER_FETCH_COMPLETED,
-				purchases: data,
-				userId,
-			} );
-		} )
-		.catch( () => {
-			dispatch( {
-				type: PURCHASES_USER_FETCH_FAILED,
-				error: PURCHASES_FETCH_ERROR_MESSAGE,
-			} );
-		} );
-};
+export const fetchUserPurchases = userId => ( { type: PURCHASES_USER_FETCH, userId } );
```

Comment:
This action now includes `userId`, which allows data-layer success/failure handlers to distpach with that data.